File: 백준/Gold/1253. 좋다/좋다.py
n = int(input())
a = list(map(int, input().split()))
a.sort()
result = 0

for k in range(n):
    find = a[k]
    i = 0
    j = n-1
    while i < j:
        if a[i]+a[j] == find:
            if i != k and j != k:
                result += 1
                break
            elif i == k:
                i += 1
            else:
                j -= 1
        elif a[i]+a[j] < find:
            i += 1
        else:
            j -= 1

print(result)

File: 백준/Gold/1253. 좋다/좋다.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
// 투포인터
    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] A = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(A);

        int result = 0;
        for (int k = 0; k < N; k++) {
            int i = 0, j = N - 1;
            while (i < j) {
                if (A[i] + A[j] == A[k]) {
                    if (i != k && j != k) {
                        result++;
                        break;
                    } else if (i == k) {
                        i++;
                    } else if (j == k) {
                        j--;
                    }
                } else if (A[i] + A[j] < A[k]) {
                    i++;
                } else {
                    j--;
                }
            }
        }

        System.out.println(result);

    }
// 음수도 숫자다!
}